### ISSUE TITLE
About cities' locations

### ISSUE THREAD

**Issue by derHochmeister** [Idhrendur/paradoxGameConverters#358]
I managed to compile a list of province ID-city from vanilla HoI4 and the More VPs mod.

https://pastebin.com/MAadaXAi

It'll take a lot of time to confront it with the province mappings.

**Comment by Idhrendur** [comment]
It might be a good practice program for me to write in Python:

* Import the province mappings  
* Import this list
* For every city in this list, check that there's a corresponding mapping
* For every city in this list, verify that it's the first linked HoI4 province in the relevant mapping

That'll give us hints for improved province mappings.

**Comment by IohannesIohannium** [comment]
I agree, but I think it'd be better if we also solve this first. https://github.com/Idhrendur/paradoxGameConverters/issues/383

**Comment by Idhrendur** [comment]
I wrote a script that compares against province mappings, and got these results:
[output.txt](https://github.com/Idhrendur/paradoxGameConverters/files/1571971/output.txt)

**Comment by Idhrendur** [comment]
I updated the script to output the names according to both files, as there are cases where a city we don't care about is in the city locations list. @IohannesIohannium, would you mind using this results file to strip those cities from the original list? The very first entry is a good example. The Vic2 province (and thus the converted city) that matters is Umeå, but the city locations list includes Storuman, which is also part of that mapping, and thus it reports a problem.

[output.txt](https://github.com/Idhrendur/paradoxGameConverters/files/1572095/output.txt)

**Comment by Idhrendur** [comment]
Update: use this one instead, which doesn't include the cases where the names match exactly. We know we want to address those.

[output-non matching names.txt](https://github.com/Idhrendur/paradoxGameConverters/files/1574023/output-non.matching.names.txt)

**Comment by Idhrendur** [comment]
One more update. This file should be far more clear about everything: 
[unmatched names.txt](https://github.com/Idhrendur/paradoxGameConverters/files/1577915/unmatched.names.txt)

**Comment by Idhrendur** [comment]
So, I put my script on github: https://github.com/Idhrendur/cityLocationScript

If you download it all, install python 3, add the province mappings file to the same directory, then double-click cityLocationScript.py, it'll generate a bunch of files with useful data.

I might do some updates with the resulting data eventually, but I think I'll focus on coding for a while again.

**Comment by IohannesIohannium** [comment]
Thanks for the file. I sadly know nothing of programming, but the txt you gave me will make me refine mappings in a later update.

**Comment by Idhrendur** [comment]
You shouldn't have to program anything, but if it's easier for me to just provide updated files from the script, I can do that too.

**Comment by Idhrendur** [comment]
@IohannesIohannium: Note sure if you're interested in looking at this more with the latest province mappings. If your are, here's files from the script indicating possible problems, the city locations file (some lines might just need commenting out), and the most up to date province mappings:
[badmappings.txt](https://github.com/Idhrendur/paradoxGameConverters/files/2164431/badmappings.txt)
[badpositions.txt](https://github.com/Idhrendur/paradoxGameConverters/files/2164432/badpositions.txt)
[cityLocations.txt](https://github.com/Idhrendur/paradoxGameConverters/files/2164433/cityLocations.txt)
[Uploading province_mappings.txt…]()

**Comment by IohannesIohannium** [comment]
Fixed the file as to solve badpositions.txt and most badmappings.txt:

[province_mappings.txt](https://github.com/Idhrendur/paradoxGameConverters/files/2166621/province_mappings.txt)

**Comment by Idhrendur** [comment]
Awesome! I don't know if you consider that good enough (it's pretty dang good), but if you wanted to use the script output to fix the city locations file (after which I could re-run the script to see if more drops out), the bad mappings and unmatched names indicate things my script couldn't find.
[cityLocations.txt](https://github.com/Idhrendur/paradoxGameConverters/files/2166819/cityLocations.txt)
[badmappings.txt](https://github.com/Idhrendur/paradoxGameConverters/files/2166820/badmappings.txt)
[unmatched names.zip](https://github.com/Idhrendur/paradoxGameConverters/files/2166824/unmatched.names.zip)
